### ISSUE TITLE
RuntimeError: element 0 of tensors does not require grad and does not have a grad_fn

### ISSUE THREAD

**Issue by mauricioarmani** [davidmascharka/tbd-nets#8]
Trying to reproduce the experiments on train-model.ipynb and using the proposed enviroment with pytorch 0.4.1 the code produced the following error:

```
RuntimeError                              Traceback (most recent call last)
<ipython-input-14-82ec354902a5> in <module>()
      6     epoch += 1
      7     print('starting epoch', epoch)
----> 8     train_epoch()
      9 
     10 save_checkpoint(epoch, 'example-{:02d}.pt'.format(epoch))

<ipython-input-13-2216c33e0bef> in train_epoch()
     33 
     34         loss_file.write('Loss: {}\n'.format(loss.item()))
---> 35         loss.backward()
     36         optimizer.step()
     37         break

~/anaconda2/envs/tbd-env/lib/python3.6/site-packages/torch/tensor.py in backward(self, gradient, retain_graph, create_graph)
     91                 products. Defaults to ``False``.
     92         """
---> 93         torch.autograd.backward(self, gradient, retain_graph, create_graph)
     94 
     95     def register_hook(self, hook):

~/anaconda2/envs/tbd-env/lib/python3.6/site-packages/torch/autograd/__init__.py in backward(tensors, grad_tensors, retain_graph, create_graph, grad_variables)
     88     Variable._execution_engine.run_backward(
     89         tensors, grad_tensors, retain_graph, create_graph,
---> 90         allow_unreachable=True)  # allow_unreachable flag
     91 
     92 

RuntimeError: element 0 of tensors does not require grad and does not have a grad_fn
```
Pytorch is trying to backpropagate through a tensor with no grad_fn, but I wasn't able to find the problem yet.

**Comment by davidmascharka** [comment]
Have you modified the notebook? I've just run the training notebook with PyTorch 0.5a and it ran successfully with just a few warnings. Are you using the hdf5 or the npy files for training?

Note: Edited your issue for code-block formatting.

**Comment by davidmascharka** [comment]
Whoops, our validation code ran `torch.set_grad_enabled(False)` and it was never set to true in the training loop. 9de0235 should fix this. Can you pull and verify?

**Comment by mauricioarmani** [comment]
Thank you, @davidmascharka. Issue solved with 9de0235a0ec42fd675a185f88057a1f98d176c4a.
